teams.md

```diff
@@ -0,0 +1,208 @@
+# Teams
+
+This document describes the responsibilities, permissions and criteria for membership
+for each of the teams associated with the _pandas_ project.
+
+### Core Library Team
+
+#### Responsibilities
+
+* Maintain and develop the _pandas _library (the
+  [https://github.com/pandas-dev/pandas/](https://github.com/pandas-dev/pandas/)
+  repository)
+* Review and Approve or Reject Pull Requests
+    * Current guidelines on merging PRs:
+      [https://pandas.pydata.org/docs/dev/development/maintaining.html#merging-pull-requests](https://pandas.pydata.org/docs/dev/development/maintaining.html#merging-pull-requests)
+* Build and Distribute Releases of the _pandas_ library
```

Comment:
maybe this should be the responsibility of the infrastructure team?

The core library team would then only be responsible for agreeing the features/major changes to include in the releases and managing the release from this perspective.

Reply:
If an infrastructure team exists, I would expect the builds, distribution, release dates, rollbacks etc to be within their purview, but i can see arguments for both.

Reply:
I think that would require discussion among the current core team to determine who should own the release process.  We were thinking that the infrastructure team was more responsible for CI jobs, servers used for testing performance, etc.

Reply:
I think the release process should be the responsibility of the infrastructure team whereas the release cycle and code additions of the backport branch is the responsibility of the core (library) team.

So this gives them, the infrastructure team, (or some subset of them) the authorities for PyPI and conda-forge as well as access to the documentation and web servers.

I would like us to be in a position in the future where a release manager purely coordinates the PRs for the features/bug fixes and only need to tag a commit to accomplish their role as release manager.

Reply:
Discussed on 6/5/24.  Used "Finalize releases" for core team, moved build/distribute to infrastructure.